### PR TITLE
VMware: Add support for storagePod

### LINKED ISSUE #57849: VMWare: OVF deploy is not supporting StoragePod
##### SUMMARY

I trying to deploy an OVA file to a StoragePod, but seems like is not supported in the vmware_deploy_ovf module, something that the vmware_guest module support when deploy a VM from a template.
The function find_datastore_by_name() used in there is just looking for 'Datastore' and not taking into account the 'StoragePod' object.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
vmware_deploy_ovf

##### ANSIBLE VERSION

```
ansible 2.7.0
  config file = /home/devel/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.6/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.6.8 (default, Apr  8 2019, 18:17:52) [GCC 8.3.0

```

##### CONFIGURATION

```
DEFAULT_LOG_PATH(/home/devel/ansible.cfg) = /tmp/ansible.log
```

##### OS / ENVIRONMENT

Running on Alpine Docker container

##### STEPS TO REPRODUCE

```
- hosts: all
  gather_facts: false
  connection: local

  vars_prompt:
    - name: "vcenter_password"
      prompt: "vSphere Password"

  tasks:
  # get date
  - set_fact: creationdate="{{lookup('pipe','date "+%Y/%m/%d %H:%M"')}}"

  - name: Deploy OVA
    vmware_deploy_ovf:
      hostname: "{{ vcenter_host }}"
      username: "{{ vcenter_username }}"
      password: "{{ vcenter_password }}"
      validate_certs: no
      cluster: Developer
      networks: "{u'VM Network':u'{{  'VLAN 89' }}'}" 
      datacenter: "{{ vcenter_dc }}"
      folder: "{{ vcenter_dc }}/vm/{{ vcenter_folder }}"
      datastore: "{{ vcenter_ds }}"
      name: "{{ vcenter_vm_name }}"
      power_on: yes
      wait: no
      ovf: "{{ ova_file }}"
```
{{ vcenter_ds }} is pointing to StoragePod object

##### EXPECTED RESULTS

OVA deployed into the StoragePod

##### ACTUAL RESULTS

```paste below
FAILED! => {"changed": false, "msg": "XXXXX could not be located"}
```

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Now, user can be specify datastore cluster for deploying OVF

Fixes: #57849 #51826

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
lib/ansible/module_utils/vmware.py
lib/ansible/modules/cloud/vmware/vmware_deploy_ovf.py

### CODE REVIEW

File: lib/ansible/modules/cloud/vmware/vmware_deploy_ovf.py [hunk 1]
```diff
@@ -309,22 +319,38 @@ def __init__(self, module):
...
 
+        self.datastore = None
+        datastore_cluster_obj = self.find_datastore_cluster_by_name(self.params['datastore'])
+        if datastore_cluster_obj:
+            datastore = None
+            datastore_freespace = 0
+            for ds in datastore_cluster_obj.childEntity:
+                if isinstance(ds, vim.Datastore) and ds.summary.freeSpace > datastore_freespace:
+                    # If datastore field is provided, filter destination datastores
+                    if ds.summary.maintenanceMode != 'normal':
```
Comment (mcasasola) [inline lib/ansible/modules/cloud/vmware/vmware_deploy_ovf.py]: what about to check ds.summary.accessible flag here as well?

Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_deploy_ovf.py]: OK. I will add this check as well.

**Comment by Akasurde** [comment]
@mcasasola Thanks for the review, could you please review this again with your suggestion ? Thanks in advance.

**Review by jillr — APPROVED** [review]

**Comment by Akasurde** [comment]
@Tomorrow9 I brought your code here. Could you please review this ? Thanks.

**Pull request merged**